actors/evm/src/interpreter/system.rs

```diff
@@ -109,14 +113,21 @@ impl<'r, BS: Blockstore, RT: Runtime<BS>> System<'r, BS, RT> {
         params: RawBytes,
         value: TokenAmount,
     ) -> Result<RawBytes, ActorError> {
-        self.flush()?;
+        if !self.readonly {
+            self.flush()?;
+        }
         let result = self.rt.send(to, method, params, value)?;
-        self.reload()?;
+        if !self.readonly {
```

Comment:
Let's move this into `reload` itself. If we're in read-only mode, reload should just shortcut.

Reply:
done.